app/models/Post.rb

```diff
@@ -1,19 +1,21 @@
 class Post < ApplicationRecord
+  has_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
   belongs_to :author, class_name: 'User'
-  has_many :comments
-  has_many :likes
 
-  validates :title, presence: true, length: { maximum: 250 }
-  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
-  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  after_save :update_posts_counter
 
-  after_create :update_post_counter
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :likes_counter, numericality: { only_integer: true }, comparison: { greater_than_or_equal_to: 0 }
+  validates :comments_counter, numericality: { only_integer: true }, comparison: { greater_than_or_equal_to: 0 }
```

Comment:
- [ ] **[OPTIONAL]** I highly recommend to implement the same validation for these attributes as suggested above.